.eslintrc.js

```diff
@@ -71,6 +71,17 @@ module.exports = {
   rules: {
     ...baseline.rules,
     'import/prefer-default-export': 'off',
+    '@typescript-eslint/no-unused-vars': [
```

Comment:
Why?

Reply:
It was unhappy with my `_, __, ___` variable names for filters. It's also going to apply to `for (const _ in b) { ... }` in the scroll PR. Prefix for unused variables is the default convention, in the absence of https://github.com/microsoft/TypeScript/issues/29202.

Reply:
Couldn't we just skip unused arguments, since there's no overhead according to https://github.com/mui/mui-x/pull/9120#discussion_r1214627262?

Reply:
Yeah, in some cases. But there is also this case:
```typescript
  return (_, row, column, apiRef): boolean => {
    const columnValue = apiRef.current.getRowFormattedValue(row, column);
    return columnValue != null ? filterRegex.test(columnValue.toString()) : false;
  };
```

And allowing `_` to mean an intentionally unused variable matches the standard typescript LSP behavior, for example if I "unuse" the `row` variable but prefix it with an underscore, the TS LSP understands it as an unused variable:
 
![Screenshot from 2023-06-12 10-58-12](https://github.com/mui/mui-x/assets/1423607/72d1327d-e799-4827-95ff-54eb9d226748)

![image](https://github.com/mui/mui-x/assets/1423607/878d348f-f411-416e-a3d9-547b47d5a85d)

Reply:
Alternatively, I can keep adding `/* eslint-disable @typescript-eslint/no-unused-vars */` where appropriate.